core/src/main/java/org/frankframework/logging/IbisMaskingLayout.java

```diff
@@ -58,12 +58,12 @@ public abstract class IbisMaskingLayout extends AbstractStringLayout {
 	 * Set of regex strings to hide locally, meaning for specific threads/adapters.
 	 * This is required to be set for each thread individually.
 	 */
-	private static final ThreadLocal<Set<String>> threadLocalReplace = new ThreadLocal<>();
+	private static final ThreadLocal<Map<String, Pattern>> threadLocalReplace = new ThreadLocal<>();
```

Comment:
Perhaps this is too much for now, but why not refactor this to a Set<Pattern>? That way you can only access the compiled Patterns via the ThreadLocal

Reply:
Let me see -- I thought that under some circumstances, we still wanted to be able to look at the original regexes or be able to remove by original regex. Although that method is perhaps never actually used.

Reply:
I looked at it, and I think this is a good suggestion. I will update this.